### ISSUE TITLE
Provide a positive feedback about client->agent->collector connectivity

### ISSUE THREAD

**Issue by jpkrohling** [jaegertracing/jaeger#1007]
## Problem - what in Jaeger blocks you from solving the requirement?
Right now, it's hard to find out if a span reported by the client will arrive at the UI at all. To figure out if the connectivity is working, we ask users to get the client to talk to the collector directly, via HTTP.

## Proposal - what do you suggest to solve the problem or improve the existing situation?
The agent has a set of endpoints on port 5778 for sampling and baggage managers. The initial proposal would be to add also a "ping" endpoint that clients can call when they are bootstrapped. Clients can then log a message if they can't reach this endpoint.

A similar endpoint can be added to the collector, to be called either by the client or by the agent.

A more elaborate setup would be to send a specially crafted span to the agent and the agent would then provide the client with a positive confirmation that it has seen this span. As the agent is mostly stateless, this approach would cause a bigger impact, so, I think we should start with the simple solution first and move to this one if needed.

**Comment by yurishkuro** [comment]
We could have a flag in the clients to verify availability of the agent, eg its config http port. Alternatively, to avoid changing all clients, we could have the same as a small test utility, a probe, that they can run in the same way as their other applications to check the setup.

**Comment by jpkrohling** [comment]
We can have it all: we could change the clients on demand *and* have the utility you described :-)

**Comment by vprithvi** [comment]
I agree that having a feedback loop that provides confidence that the client -> agent -> collector connectivity is important. 

I like your idea of having the client verify the availability of the agent by means of a rpc, which would allow service owners to quickly rectify any configuration issues. Another alternative would be to move to a connection based protocol for span submission (which would error when the first span is flushed)

See related discussion here: https://github.com/jaegertracing/jaeger-client-java/issues/408

**Comment by pavolloffay** [comment]
We should also reconsider logging failed sends in clients. At least in java the default level is debug.

**Comment by annanay25** [comment]
As per @isaachier's debug span solution, this issue is blocked on #1079.

**Comment by yurishkuro** [comment]
won't fix, this part of ingestion is now mostly covered by OTEL SDKs and receivers.
